### PR TITLE
EC2 monthly report

### PR DESCRIPTION
**Pull request by simonmeyerrr**
This PR adds:
### - One more subtask in the task `process-account`
This task get EC2 billing data in ES and then get for each instances present in billing data.
Only one report is produced every month for the previous month.

In the future, we will be able to add other products for this report like RDS.

### - An endpoint to get this data `/ec2/history`
This endpoint is the same as `/ec2` but get the data of a specific month
Query args:
- `accounts`: list of accounts separated by a `,` (Optionnal)
- `date`: date of the report we want to get (have to be the first day of a month ex: `2018-08-01`

### - Correction for the cost of daily ec2 report

### CODE REVIEW

File: aws/history/history.go [hunk 1]
```diff
@@ -0,0 +1,222 @@
...
+		case "AmazonCloudWatch":
+			cloudWatchCost = product.Instances
+			break
+		}
+	}
+	err := getEc2HistoryReport(ctx, ec2Cost, cloudWatchCost, aa, startDate, endDate)
+	return err
+}
+
+// FetchHistoryInfos fetchs billing data et stats of EC2 instances of the last month (only EC2 for now=
```
Comment (Lolincolc) [inline aws/history/history.go]: "et" -> "and"

File: aws/history/history_ec2.go [hunk 2]
```diff
@@ -0,0 +1,399 @@
...
+					IOWrite:    stats.IOWrite,
+					Cost:       inst.Cost,
+				}
+			}
+		}
+	}
+	return nil
+}
+
+// getEc2Metrics get credentials, accounts and region to fetch EC2 instances stats
```
Comment (Lolincolc) [inline aws/history/history_ec2.go]: "get" -> "get**s**"

File: aws/history/history_ec2.go [hunk 3]
```diff
@@ -0,0 +1,399 @@
...
+		Do(context.Background()); err != nil {
+		logger.Error("Error when putting InstanceInfo history in ES", err.Error())
+		return err
+	} else {
+		logger.Info("Instance put in ES", *res)
+		return nil
+	}
+}
+
+// filterEc2Instances filter instances and volumes of EC2 instances
```
Comment (Lolincolc) [inline aws/history/history_ec2.go]: "filter" -> "filter**s**"

File: ec2/prepare_response.go [hunk 4]
```diff
@@ -127,21 +149,28 @@ func prepareResponse(ctx context.Context, resEc2 *elastic.SearchResult, user use
...
+	if err != nil {
+		return nil, err
+	}
+	for _, account := range response.TopReports.Buckets {
+		if len(account.TopReportsHits.Hits.Hits) > 0 {
+			reports = append(reports, account.TopReportsHits.Hits.Hits[0].Source)
+		}
+	}
+	return reports, nil
+}
```
Comment (Lolincolc) [inline ec2/prepare_response.go]: You forget to add a line

**Review by Lolincolc — APPROVED** [review]
Tested and approved !

**Review by hug33k — APPROVED** [review]

**Pull request merged**